Fix demo-node-todo-v2

sdemura opened the pull request:
Addresses ENGT-2714.

Fixes odd issues with with restart policies when there is no HTTP response before exit(1).

Behavior pre and post 425e and apc 0.17.2/0.17.3 is now consistent.

1. Hit the Exit button in the app, 1 instance is killed and restarted.
2. Hit the Exit button again, a second instance is killed and restarted.
3. Hit the Exit button again within 5 minutes and the job is now flapping. The killed instance will wait 5 minutes before being restarted. 

@emil2k @juanman2

emil2k commented:
Have you tested this to make sure it provides a HTTP response before the process exits? If not so, launch the app, request the quit, check the response, and then repeat like 10 times to make sure you didn't introduce a race somewhere.

emil2k commented on demo-node-todo-v2/app.js:
```diff
@@ -22,6 +22,15 @@ var server = app.listen(port, function() {
 
+function exitApp( exitNow, res ){
+    res.send("Responding before exit.");
+    if (exitNow == true) { // reset back to true
```
I don't understand your use of `exitNow` it seems useless.

sdemura replied:
exitNow is there to ensure that when the page is refreshed from an exit that the next instance doesn't get exit itself. When you click the exit button, set exitNow to true, which will kick off the quit function. Inside that function set it back to true.

NodeJS and its asynchronous programming is new to me, so there may be a better way...

emil2k commented on demo-node-todo-v2/app.js:
```diff
@@ -36,9 +45,10 @@ app.get('/base.css', routes.base);
-  process.exit(1);
-});
+app.get('/quit', function(req, res){
+        exitApp(true, res);
```
This essentially simplifies to:

```javascript
    res.send("Responding before exit.");
    console.log("Exit button clicked. Attempting to kill process.")
    process.exit(1);
```

There is no guarantee or even time allotted for response to return before the process exits, I doubt this works.

sdemura commented:
Committed prematurely. Rebased and fixed. 

Updated code:

```function exitApp( exitNow, res ){
    if (exitNow == true) {
        exitNow = false;
        console.log("Exit button clicked. Attempting to kill process with exit(1).");
        res.writeHead(200);
        res.end("goodbye, world!")
        process.exit(1);
    }
}
```

Respond with 200, then exit. Tested this multiple times on old and new cluster versions.

emil2k commented:
Hold off on merging, I'm going to test this app myself, when I get to the office.

By the way, this is again stated incorrectly: "... odd issues with with restart policies". This is regarding routing behavior and they are not "issues" as was repeatedly explained to you.

emil2k commented:
LGTM, tested app with restarting.

sdemura commented:
Merged #28
